backend/models/user.js

```diff
@@ -139,17 +179,28 @@ User.authenticate =  async function (logger, username, password) {
 
 		user.customData.lastLoginAt = new Date();
 
-		await User.update(username, {"customData.lastLoginAt": user.customData.lastLoginAt});
+		await db.update("admin", COLL_NAME, {user: username}, {$set: {"customData.lastLoginAt": user.customData.lastLoginAt}});
 
 		logger.logInfo("User has logged in", {username});
 
 		return user;
 	} catch(err) {
+		let resCodeError = err.resCode ? err : { resCode: utils.mongoErrorToResCode(err) };
```

Comment:
can we move the try/catch?

We're catching exceptions that we are throwing here unnecessarily.

Effectively the catch should only be around the authentication part. Then you wouldn't have to probe the resCode value just to check if the user failed to authenticate itself, or or check if user exists etc.

i.e. this whole blurb can simplify if (err.resCode) { //deal with account lockout code } otherwise, throw mongo error